execution.bs

```diff
@@ -4973,15 +5088,79 @@ enum class forward_progress_guarantee {
+            [dcl.struct.bind].</span>
+
+        4. The member <code><i>default-impls</i>::<i>get-attrs</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              [](const auto& data, const auto&... child) noexcept -> decltype(auto) {
+                if constexpr (sizeof...(child) == 1)
+                  return <i>FWD-ENV</i>(execution::get_env(child...)); //
+                else
+                  return empty_env();
+              }
+              </pre>
+
+        5. The member <code><i>default-impls</i>::<i>get-env</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Rcvr>(auto index, auto& state, const Rcvr& rcvr) noexcept
+                -> decltype(<i>FWD-ENV</i>(execution::get_env(rcvr))) {
+                return <i>FWD-ENV</i>(execution::get_env(rcvr));
+              }
+              </pre>
+
+        6. The member <code><i>default-impls</i>::<i>get-state</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Sndr>(Sndr&& sndr, auto& rcvr) noexcept -> decltype(auto) {
+                return get&lt;1>(std::forward&lt;Sndr>(sndr));
+              }
+              </pre>
+
+        7. The member <code><i>default-impls</i>::<i>start</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              [](auto& state, auto& rcvr, auto&... ops) noexcept -> completion_signatures<> {
+                (execution::start(ops), ...);
+                return {};
+              }
+              </pre>
+
+        8. The member <code><i>default-impls</i>::<i>complete</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Index, class Rcvr, class Tag, class... Args>(
+                Index, auto& state, Rcvr& rcvr, Tag, Args&&... args) noexcept
+                  -> completion_signatures&lt;Tag(Args...)>
+                    requires <i>callable</i>&lt;Tag, Rcvr, Args...> {
+                <i>// Mandates: Index::value == 0</i>
+                tag(std::move(rcvr), std::forward&lt;Args>(args)...); 
+                return {};
+              }
+              </pre>
+
+        9. The return type of <code><i>basic-sender</i></code>'s customization of
+            `get_completion_signatures` is computed as follows.
 
-            - The unspecified base type has no non-static data members. It may
-                define member functions or hidden friend functions ([hidden.friends]).
+              1. Let `Env` be the type of an environment, let the type `Sndr` be
+                  the (possibly cv- and ref-qualified) type of a specialization
```

Comment:
"ref-qualifier" only applies to member functions, so I don't think we can say "ref-qualified" to mean a reference type. And in any case, if `Sndr` is `const S` or `S&` then it's _not_ a specialization of `basic_sender`.

How about this:

```suggestion
                  a type such that `remove_cvref_t<Sndr>` is a specialization
```